lib/puppet/provider/elb_loadbalancer/v2.rb

```diff
@@ -298,7 +298,7 @@ def update_listeners
     # which to add, and which need modification.  The modification is a bit of
     # a misnomer, since it really ends up being a delete followed by an add.
 
-    is_listener_ports = @property_hash[:listeners].collect {|x| x['load_balancer_port'].to_i }
+    is_listener_ports = Array(@property_hash[:listeners]).collect {|x| x['load_balancer_port'].to_i }
```

Comment:
More fun, just because:
```ruby
[1] pry(main)> a = {a: {a: 1}}
=> {:a=>{:a=>1}}
[2] pry(main)> Array(a)
=> [[:a, {:a=>1}]]
[3] pry(main)> a = [{a: 1}]
=> [{:a=>1}]
[4] pry(main)> Array(a)
=> [{:a=>1}]
```